desktop/cmp/grid/ag-grid/ColumnHeader.js

```diff
@@ -0,0 +1,132 @@
+export class ColumnHeader extends Component {
+
+    //------------------------
+    // Immutable public properties
+    //------------------------
+    gridModel;
+    column;
+    colId;
+
+    menuButton = new Ref();
+
+    // Get any active sortBy for this column, or null
+    @computed
+    get activeGridSorter() {
+        return this.gridModel.sortBy.find(it => {
+            return it.colId === this.colId;
+        });
+    }
+
+    constructor(props) {
+        super(props);
+        const {gridModel, column} = this.props;
+        this.gridModel = gridModel;
+        this.column = column;
+        this.colId = column.colId;
+    }
+
+    render() {
+        const {displayName} = this.props;
+
+        return div({
+            className: 'xh-grid-header',
+            onClick: this.onClick,
+            items: [
+                span(displayName),
+                this.renderSortIcon(),
+                this.renderMenuIcon()
+            ]
+        });
+    }
+
+    renderSortIcon() {
+        const {props, activeGridSorter} = this,
+            {enableSorting} = props;
+
+        if (!enableSorting || !activeGridSorter) return null;
+
+        if (activeGridSorter.abs) {
+            return Icon.arrowToBottom();
+        } else if (activeGridSorter.sort === 'asc') {
+            return Icon.arrowUp({size: 'sm'});
+        } else if (activeGridSorter.sort === 'desc') {
+            return Icon.arrowDown({size: 'sm'});
+        }
+        return null;
+    }
+
+    renderMenuIcon() {
+        const {enableMenu, showColumnMenu} = this.props;
+        if (!enableMenu) return null;
+
+        return div({
+            item: Icon.bars(),
+            ref: this.menuButton.ref,
+            onClick: (e) => {
+                e.stopPropagation();
+                showColumnMenu(this.menuButton.value);
+            }
+        });
+    }
+
+    //--------------------
+    // Implementation
+    //--------------------
+    onClick = (e) => {
+        if (!this.props.enableSorting) return;
+
+        const {gridModel, activeGridSorter, colId} = this,
+            nextSortBy = this.getNextSortBy();
+
+        // Add to existing sorters if holding shift, else replace
+        let sortBy = e.shiftKey ? clone(gridModel.sortBy) : [];
+        if (activeGridSorter) {
+            sortBy = sortBy.filter(it => it.colId !== colId);
+        }
+        if (nextSortBy) {
+            sortBy.push(nextSortBy);
+        }
+
+        this.gridModel.setSortBy(sortBy);
+    }
+
+    getNextSortBy() {
+        const {colId, column, activeGridSorter} = this,
+            {sort, abs = false} = activeGridSorter || {};
+
+        if (!activeGridSorter) {
+            return {colId, sort: 'asc', abs: false};
+        } else if (sort === 'asc') {
+            return {colId, sort: 'desc', abs: false};
+        } else if (sort === 'desc' && column.absSort && !abs) {
+            return {colId, sort: 'desc', abs: true};
+        } else {
+            return null;
```

Comment:
This removes the sort when you have a non-abs col that was desc.  Wouldn't it be more natural to toggle that back to asc?

Reply:
Sorry - just realized that is the default ag-grid behavior. Now not sure how standard that is - I found it odd and don't think most grids/tables work that way....  

"Unsorted" is a bit of a funny term too - assuming this is the order the data was provided to the store. But from a use perspective imagine the data happened to be name alpha asc. Let's also assume we've sorted on some other value by default.  Clicking the name header once would sort asc, again would sort dec, again would appear to sort asc but in a subtle way be "no sort" and then the fourth click would appear to do nothing...  

I dunno - not eager to change ag-grid functionality but found this weird.  E.g. see MacOS finder columns...